Sources/TensorFlow/LazyTensor/LazyTensorOperation.swift

```diff
@@ -0,0 +1,551 @@
+import CTensorFlow
+
+/// The `TF_Tensor *` type.
+typealias CTFTensor = OpaquePointer
+
+public class LazyTensor : _AnyTensorHandle {
+    enum Handle {
+        /// Bool indicates if this concrete TFETensorhandle was a result of
+        /// materialization.
+       case conc(TFETensorHandle, Bool)
+        /// Bool indicates whether this is held by a swift variable,
+        /// i.e., is this a live tensor?
+       case sym(LazyTensorOperation, Int, Bool)
+    }
+
+    let handle: Handle
+
+    public var _tfeTensorHandle: TFETensorHandle {
+        switch (handle) {
+        case Handle.conc(let h, _):
+            return h
+        case Handle.sym(let op, let index, _):
+            assert(false, "TODO: to be send out in a separate PR.")
+            // return op.materialized(index: index)
+        }
+    }
+
+    init(_ base: TFETensorHandle) {
+        handle = Handle.conc(base, false)
+    }
+
+    init(_materialized base: TFETensorHandle) {
+        handle = Handle.conc(base, true)
+    }
+
+    init(_lazy op: LazyTensorOperation, index: Int) {
+        precondition(index < op.outputCount, "Symbolic Tensor Index is out-of-bounds")
+        handle = Handle.sym(op, index, false)
+    }
+
+    init(_lazyLive op: LazyTensorOperation, index: Int) {
+        precondition(index < op.outputCount, "Symbolic Tensor Index is out-of-bounds")
+        handle = Handle.sym(op, index, true)
+    }
+}
+
+public class LazyTensorOperation : TensorOperation {
```

Comment:
Is there a reason this type needs to be a public API?

Reply:
Actually no. Removed `public` from class and all APIs.